Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py

```diff
@@ -0,0 +1,823 @@
+            center = np.mean(self.data_2d[np.where(self.model.labels_ == cluster_)], axis=0)  # type: ignore
+            self.centers[cluster_] = center
+
+
+class PostProcessing(object):
+    """
+    Class to analyze the clustering
+    """
+
+    def __init__(self, clustering: Type[Clustering], threshold: float, generic_cluster_name: bool):
+        """
+        Instantiate class object for visualization
+        :param clustering: Object Clustering
+        :param threshold: Threshold for the cluster homogeneity
+        :param generic_cluster_name: Boolean if cluster don't have name and needs generic naming
+        :return: Instantiate class object for visualization
+        """
+        self.clustering = clustering  # type: Type[Clustering]
+        self.threshold = threshold  # type: float
+        self.generic_cluster_name = generic_cluster_name
+        self.stats = {}  # type: ignore
+        self.statistics()
+        self.compute_dist()
+        self.date_training = datetime.now().strftime("%m/%d/%Y %H:%M:%S")
+        self.summary = None  # type: ignore
+        self.json = None  # type: ignore
+
+    def statistics(self):
+        """
+        Compute statistics of the clusters
+        """
+        # plot_silhouette = self.com_silhouette()
+        self.stats['General'] = {}
+        self.stats['General']['Nb sample'] = self.clustering.raw_data.shape[0]  # type: ignore
+        self.stats['General']['Nb cluster'] = self.clustering.number_clusters
+        self.stats['General']['min_samples'] = self.clustering.model.min_samples  # type: ignore
+        self.stats['General']['min_cluster_size'] = self.clustering.model.min_cluster_size  # type: ignore
+        for number_cluster in range(0, self.clustering.number_clusters):  # type: ignore
+            self.stats[number_cluster] = {}
+            self.stats[number_cluster]['number_samples'] = sum(
+                self.clustering.model.labels_ == number_cluster)  # type: ignore
+            ind = np.where(self.clustering.model.labels_ == number_cluster)[0]  # type: ignore
+            selected_data = [x for x in self.clustering.raw_data.iloc[ind][FAMILY_COLUMN_NAME]]  # type: ignore
+            counter = collections.Counter(selected_data)
+            total = sum(dict(counter).values(), 0.0)
+            dist = {k: v * 100 / total for k, v in counter.items()}
+            dist = dict((k, v) for k, v in dist.items() if v >= 1)
+            self.stats[number_cluster]['distribution sample'] = dist
+
+    def compute_dist(self):
+        """
+        Compute distribution of sample per cluster (depending of the naming and threshold)
+        """
+        dist_total = {}  # type: Dict
+        if not self.generic_cluster_name:
+            for cluster_number in range(0, self.clustering.number_clusters):  # type: ignore
+                chosen = {k: v for k, v in self.stats[cluster_number]['distribution sample'].items() if
```

Comment:
the logic inside the if and else looks almost the same (of the loop)
maybe we can ccreate a function? or do some condition inside the loop?

Reply:
Yes but then we will have multiple if in the loop (more than 4).